### PR TITLE
Add docs for standards

### PR DESCRIPTION
**Pull request by alcatrazEscapee**
Wrote the standards into two separate documents:

- [Rendered View](https://github.com/alcatrazEscapee/mappings-testing/blob/main/docs/standards.md) `standards.md` wraps the Parameter name / javadoc sections from the temp. working doc. Some minor wording was changed to target the document towards the audience of "people who want to submit mappings"
- [Rendered View](https://github.com/alcatrazEscapee/mappings-testing/blob/main/docs/review-merge-process.md) `review-merge-process.md` wraps the "Review / Merge process" from the temp working doc. (*real.. real shocker that one*). It's intended as a codification of the review and merge standards and procedures the team will adhere to and is worded as such.

I intentionally excluded the "Quality of Life tools" section as that's basically just brainstorming and dreams, and the CLA, because, well, it's already in the CLA bot.

### CODE REVIEW

**Comment by ChampionAsh5357** [comment]
I would make these ordered lists so we can refer to them on the fly (like if someone violates use american english names for spelling, I could say violates Params 1 instead of the actual requirement).

File: docs/standards.md
Comment (ChampionAsh5357) [inline docs/standards.md]: Doesn't 4 and 6 contradict each other? Were they both in the doc and we just forgot removing one or the other?

Comment (alcatrazEscapee) [reply docs/standards.md]: They were both in the doc yes. I would agree they're not perfectly worded but disagree they're contradictory. i.e. `Level level` is both verbose (i.e. not `lvl`), a complete word, and based off the class names. Although I'm not sure how to reword these exactly to be more clear

Comment (ChampionAsh5357) [reply docs/standards.md]: Combine them into the same point then. `Parameter names should be based off of the class names and the current context. They should be verbose and valid words, differentiating when there are multiple of the same type.`

File: docs/standards.md
Comment (ChampionAsh5357) [inline docs/standards.md]: Just realized this parameter list itself is invalid if within a method signature. Want to avoid any ambiguity that might come or would result in a later PR of people wondering the same.

File: docs/standards.md
Comment (ChampionAsh5357) [inline docs/standards.md]: Based on https://github.com/ParchmentMC/mappings-testing/pull/1#discussion_r645109509, we should adjust the content information to include the same.

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]

**Review by ChampionAsh5357 — APPROVED** [review]

**Review by SizableShrimp — CHANGES_REQUESTED** [review]
The content looks good to me. However, I might suggest some style changes to the general markdown formatting, e.g. using headers/labels over colons. Also moving `review-merge-process.md` to use an ordered list as well. Note that you can repeat `1.` over and over and still get them numbered. Might wanna do that to make adding new entries to the list easier later down the line, but personal preference. Otherwise seems good to merge.

File: docs/review-merge-process.md
Comment (sciwhiz12) [inline docs/review-merge-process.md]: This could be narrowed down to 'from a reviewer from the Parchment mapping reviewers team.'. Also, should this also note that reviews from the author do not count towards the two-review requirement?

File: docs/review-merge-process.md
Comment (sciwhiz12) [inline docs/review-merge-process.md]: This sorta has a conflict with the `A PR requires two approving reviews; one must be from a Parchment team member` point: Who will the second review be from, if not from a Parchment team member? Do external reviews count towards the two-approvals requirement?

File: docs/review-merge-process.md
Comment (sciwhiz12) [inline docs/review-merge-process.md]: What does "all reviewers" mean in this context? Because the first point specifies that a simple majority of all Parchment team reviewers, then it follows that the second point must be wider in scope than "all Parchment team reviewers". 
 - Does this mean that external (non mapping-team) reviewers count? 
 - Does this mean that all mapping team members (not just those in the reviewers subteam) are counted? 
 - What happens if we reasonably can't get the votes of a majority of the reviewers due to lack of activity on their part?

We should outline this process a bit more in a separate document (whether in this PR or in a future one).

File: docs/review-merge-process.md
Comment (sciwhiz12) [inline docs/review-merge-process.md]: Should we have a small delay between the last required approval reivew and the actual merge? This would give both the reviewers team and external reviewers a window of time to confirm that nothing is majorly out of place.

File: docs/standards.md
Comment (sciwhiz12) [inline docs/standards.md]: Point 7 is also covered by Point 8 (as the regex does not include `$` and `_`).
Also, there should be a written explanation of the regex (in a subpoint), for people not fluent in regex.
> Parameters must start with the `p` prefix, followed by an uppercase letter, followed by one or more letters or numbers. Any other characters (such as `$` or `_` are not allowed).

File: docs/standards.md
Comment (sciwhiz12) [inline docs/standards.md]: Is this a suggestion to use Mojmap/Official class names, or a restriction? Could we be given some examples of what exceptions may apply here (even as a comment on this PR)?

File: docs/standards.md
Comment (sciwhiz12) [inline docs/standards.md]: This should be moved as either a subpoint or the point directly after #<area>5. Also, the parameters should have the `p` prefix.

File: docs/standards.md
Comment (sciwhiz12) [inline docs/standards.md]:
```suggestion
    - Example: `{@link net.minecraft.math.BlockPos}` instead of `{@link BlockPos}`
```
Also, should this restriction only be in force for 'dumb' tags (tags which do not validate their contents), such as `{@code}`?

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
I have a _few_ questions and changes to note. :grin:

File: docs/review-merge-process.md
Comment (alcatrazEscapee) [reply docs/review-merge-process.md]: > Who will the second review be from, if not from a Parchment team member?

Anyone who wants to review it.

> Do external reviews count towards the two-approvals requirement?

Yes, that is why two are required, but only one is mentioned as required to be from Parchment.

File: docs/review-merge-process.md
Comment (alcatrazEscapee) [reply docs/review-merge-process.md]: So based on my initial assumptions:

> Does this mean that external (non mapping-team) reviewers count?
> Does this mean that all mapping team members (not just those in the reviewers subteam) are counted?

I have not made a distinction between Parchment mapping reviewers and general members of Parchment - whichever team makes more sense to be used (probably mapping reviewers, which was not fully clarified at the time this document was written), makes sense here.

> What happens if we reasonably can't get the votes of a majority of the reviewers due to lack of activity on their part?

The the status quo (PR not merged) is maintained.

File: docs/review-merge-process.md
Comment (alcatrazEscapee) [reply docs/review-merge-process.md]: Unsure. A valid question to bring up. I don't terribly mind either way.

File: docs/standards.md
Comment (alcatrazEscapee) [reply docs/standards.md]: > Also, there should be a written explanation of the regex (in a subpoint), for people not fluent in regex.

This is *why* there are individual points explaining each note not covered by the regex (i.e. lower camel case, no special characters, p prefix.). Removing those would require one to understand this regex - hence why Point 7 exists.

File: docs/standards.md
Comment (alcatrazEscapee) [reply docs/standards.md]: Example: if the mojmap name has a typo in it, or is using non-american english spelling. (But yes, can add that)

File: docs/standards.md
Comment (alcatrazEscapee) [reply docs/standards.md]: There are two things here:

1. making the links work, in an IDE, with imports that may or may not exist
1. being able to simply / automatically migrate names across class/package changes between versions.

Point 1. would indicate only particular tags (`@link`, `@see`). Point 2 would suggest all. Thoughts?

File: docs/standards.md
Comment (sciwhiz12) [reply docs/standards.md]: Hmm, looks like I didn't read the list too closely. :eyes:

Point 6 should clarfied a bit because it may be confusing if `pLowerCamelCase` fits the lowerCamelCase requirement (as if we strip the prefix, it's now in PascalCase).
The other individual points should be moved closer to the regex (grouping the naming restrictions together), or a separate block after the standards list should be added to state that the regex is a consolidation of the given points.

File: docs/review-merge-process.md
Comment (sciwhiz12) [reply docs/review-merge-process.md]: Discussion on the Discord has led to a recommendataion of a minimum of 48 hours between the last approving review required for merging, and the actual merging itself.

However, I still firmly believe that an option of a minimum of 24 hours be inlcuded, for small uncontestable submissions (such as typo fixes, or minor grammatical changes).

File: docs/review-merge-process.md
Comment (sciwhiz12) [reply docs/review-merge-process.md]: For the first point, we should clarify that "Parchment team reviewers" means those part of the @ParchmentMC/mapping-reviewers subteam, which is expected to be those team members who wish to actively review PRs (and so hold the privilege of being able to merge those PRs in), while "all reviewers" means those part of the broader @ParchmentMC/mappings team, which holds all team members who manage the mappings data (but also includes those who aren't in the mapping-reviewers team, meaning those who are not able to review PRs due to time or personal constraints).

For the second point, since the mapping-reviewers subteam should only have those willing to be active in managing PRs and reviews, it should be fine. While the first vote will only be voted upon by the mapping-reviewers, those not voting can still express their own opinion, and those voting should take these statements into account in their votes.

File: docs/review-merge-process.md
Comment (sciwhiz12) [reply docs/review-merge-process.md]: Fair enough, though discussion in Discord has given thought to allowing small contribution PRs to only require one approving review instead of at least two.

File: docs/standards.md
Comment (sciwhiz12) [reply docs/standards.md]: While the idea of automatically migrating names across class/package changes is a good one, it would be more complex to implement because names may appear in a) `{@link}` and etc. tags (those which are checked by `javadoc` and the IDE), b) `{@code}` and etc. tags (those which do not have any checking), and c) without any tag (referencing names normally).

In my opinion, we should for now only require the use of fully qualified names for `{@link}` tags, and not 'dumb' tags or normal textual references to classes. In the future, we can reconsider this point once we start implementing automatic name migration.

File: docs/standards.md
Comment (sciwhiz12) [inline docs/standards.md:25]: Would've been good to also add an example of a name with a number character, but this is fine as-is.

**Review by sciwhiz12 — APPROVED** [review]
This looks good to me, with one small comment which you may or may not care about adding (fine by me either way).

**Review by SizableShrimp — APPROVED** [review]

**Pull request merged**
